### DISCUSSION TITLE
MariaDB / MySQL service is unealthy

### DISCUSSION THREAD

**Discussion by tristanbes** [community/community#25118] [Actions]
<p>Hello,</p>
<p>I’ve tried multiple combinaison, but it seems I can’t figure out  <strong>why is mariadb not starting.</strong></p>
<p>I’ve look mutliple examples (such as<a href="https://github.com/shivammathur/setup-php/blob/master/examples/symfony-mysql.yml" rel="nofollow noopener">this one</a>) but nothing is working (including changing port to 3366);</p>
<p>The output log is :</p>
<pre><code>Status: Downloaded newer image for mariadb:10.4
docker.io/library/mariadb:10.4
/usr/bin/docker create --name 88c5e4e016ef4568bc6bf49cd2b81d61_mariadb104_4b3bb7 --label 488dfb --network github_network_674333f1aba94d138fe92f85cd90c518 --network-alias mysql -p 3306/tcp --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=10s --health-retries=10 -e GITHUB_ACTIONS=true mariadb:10.4
b62b9146f0bd1f3512327ee67357e76827c31c03d9f2fd2c9d06c3648b98ce0f
/usr/bin/docker start b62b9146f0bd1f3512327ee67357e76827c31c03d9f2fd2c9d06c3648b98ce0f
b62b9146f0bd1f3512327ee67357e76827c31c03d9f2fd2c9d06c3648b98ce0f
/usr/bin/docker ps --all --filter id=b62b9146f0bd1f3512327ee67357e76827c31c03d9f2fd2c9d06c3648b98ce0f --filter status=running --no-trunc --format "{{.ID}} {{.Status}}"
b62b9146f0bd1f3512327ee67357e76827c31c03d9f2fd2c9d06c3648b98ce0f Up Less than a second (health: starting)
/usr/bin/docker port b62b9146f0bd1f3512327ee67357e76827c31c03d9f2fd2c9d06c3648b98ce0f
3306/tcp -&gt; 0.0.0.0:32768
/usr/bin/docker inspect --format="{{if .Config.Healthcheck}}{{print .State.Health.Status}}{{end}}" b62b9146f0bd1f3512327ee67357e76827c31c03d9f2fd2c9d06c3648b98ce0f
starting
mysql service is starting, waiting 2 seconds before checking again.
/usr/bin/docker inspect --format="{{if .Config.Healthcheck}}{{print .State.Health.Status}}{{end}}" b62b9146f0bd1f3512327ee67357e76827c31c03d9f2fd2c9d06c3648b98ce0f
unhealthy
##[error]Failed to initialize, mysql service is unhealthy.

env:  
TZ: Europe/Paris
PHP\_VERSION: 7.4
NODE\_VERSION: 10.x
COMPOSER\_ALLOW\_SUPERUSER: '1' # https://getcomposer.org/doc/03-cli.md#composer-allow-superuser  
  
# Database  
MYSQL\_ROOT\_PASSWORD: symfony
MYSQL\_DATABASE: symfony

jobs:  
php:  
runs-on: ubuntu-latest
services:  
mysql:  
image: mariadb:10.4
ports:  
 - 3306/tcp
options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=10s --health-retries=10 
</code></pre>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/tristanbes">@tristanbes</a> ,</p>
<p>This is due to the top level env setting is not applied to the github services(‘docker create’ command in the execution step). Need to set in ‘<a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idservicesenv" rel="nofollow noopener">jobs.&lt;job_id&gt;.services.env</a>’ level. Code sample below:</p>
<pre><code>env:
    TZ: Europe/Paris
    PHP_VERSION: 7.4
    NODE_VERSION: 10.x
    COMPOSER_ALLOW_SUPERUSER: '1' # https://getcomposer.org/doc/03-cli.md#composer-allow-superuser
jobs:
    php:
        runs-on: ubuntu-latest
        services:
            mysql:
                image: mariadb:10.4
                env:
                  MYSQL_ROOT_PASSWORD: symfony
                  MYSQL_DATABASE: symfony
                ports:
                  - 3306/tcp
                options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=10s --health-retries=10
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/7179d22450d3c13bf34ec3063e7a0bc05c121340.png" data-download-href="/uploads/short-url/gbQX9NWIJMn45ZXhaCEuWyubdWU.png?dl=1" title="mariaDB.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181089005-baafd229-d561-4bef-b817-54d8f1a97467.png" alt="mariaDB.png" data-base62-sha1="gbQX9NWIJMn45ZXhaCEuWyubdWU" width="690" height="249" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/7/7179d22450d3c13bf34ec3063e7a0bc05c121340_2_690x249.png, https://aws1.discourse-cdn.com/github/optimized/2X/7/7179d22450d3c13bf34ec3063e7a0bc05c121340_2_1035x373.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/7/7179d22450d3c13bf34ec3063e7a0bc05c121340_2_1380x498.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/7179d22450d3c13bf34ec3063e7a0bc05c121340_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">mariaDB.png</span><span class="informations">1507×544 53 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by tristanbes** [comment]
<p>Thank you very much. I didn’t know that.</p>

**Comment by aryashreep** [comment]
<p>Hi <a class="mention" href="/weide-zhou">@weide-zhou</a> <a class="mention" href="/tristanbes">@tristanbes</a> </p>
<p>I have tried this but it is not working. Could you please check.</p>
<pre><code># This is a basic workflow to help you get started with Actions

name: Automated Drupal8 CICD

# Controls when the action will run. Triggers the workflow on push or pull request
# events but only for the master branch
on:
  push:
    branches: [master]
  pull_request:
    branches: [master]

env:
    PHP_VERSION: 7.3
    COMPOSER_ALLOW_SUPERUSER: '1' # https://getcomposer.org/doc/03-cli.md#composer-allow-superuser

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  # This workflow contains a single job called "service"
  services:
    # The type of runner that the job will run on
    runs-on: ubuntu-latest
    strategy:
      matrix:
        php_version: ['7.3']
    # Enable the mysql services
    services:
      mariadb:
        image: mariadb:latest
        ports:
          - 3306
        env:
          MYSQL_HOST: mysql
          MYSQL_USER: root
          MYSQL_PASSWORD: root
          MYSQL_DATABASE: drupal8
          MYSQL_ROOT_PASSWORD: root
        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=10s --health-retries=10

    steps:
      - name: Check out the code
        uses: actions/checkout@v2
        with:
          fetch-depth: 1

      - name: Verify MySQL connection from host
        run: sudo apt-get install -y mysql-client

      - name: Set up PHP ${{ matrix.php_version }}
        uses: shivammathur/setup-php@v2
        with:
          php-version: ${{ matrix.php_version }}

      - name: Install Composer dependencies
        run: composer install --dev

  phpunit:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        php_version: ['7.3']
    services:
      mariadb:
        image: mariadb:latest
        ports:
          - 3306
        env:
          MYSQL_HOST: mysql
          MYSQL_USER: root
          MYSQL_PASSWORD: root
          MYSQL_DATABASE: drupal8
          MYSQL_ROOT_PASSWORD: root
        options: --health-cmd="mysqladmin ping" --health-interval=5s --health-timeout=2s --health-retries=3

    steps:
    - name: Set up PHP ${{ matrix.php_version }}
      uses: shivammathur/setup-php@v2
      with:
        php-version: ${{ matrix.php_version }}

    - name: Check out the code
      uses: actions/checkout@v2
      with:
        fetch-depth: 1

    - name: Install Composer dependencies
      run: composer install --dev

    - name: Verify MySQL connection from container
      run: |
        sudo apt-get update
        sudo apt-get install -y mysql-client
        mysqladmin ping
        mysql -hmysql -uroot -proot -e "CREATE DATABASE IF NOT EXISTS drupal8;"
        mysql -hmysql -uroot -proot -e "SHOW DATABASES"
        mysql -hmysql -uroot -proot drupal8 &lt; scripts/database/dumps/drupal8.sql
        mysql -hmysql -uroot -proot -e "USE drupal8; SHOW TABLES;"
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/48498a05f2b2d0986367892e244b0c904a2b5369.jpeg" data-download-href="/uploads/short-url/ajtVXiCdtGy9DARMApS8z06F16p.jpg?dl=1" title="log.JPG" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/5535841/181089009-0451fe12-7233-40fe-8fa8-18e67e83124f.jpeg" alt="log.JPG" data-base62-sha1="ajtVXiCdtGy9DARMApS8z06F16p" width="690" height="445" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/48498a05f2b2d0986367892e244b0c904a2b5369_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">log.JPG</span><span class="informations">727×469 65.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
